mlir/lib/Transforms/Mem2Reg.cpp

```diff
@@ -106,7 +106,7 @@ struct MemorySlotPromotionInfo {
   /// its uses, it is because the defining ops of the blocking uses requested
   /// it. The defining ops therefore must also have blocking uses or be the
   /// starting point of the bloccking uses.
-  DenseMap<Operation *, SmallPtrSet<OpOperand *, 4>> userToBlockingUses;
+  llvm::MapVector<Operation *, SmallPtrSet<OpOperand *, 4>> userToBlockingUses;
```

Comment:
Any chance we can introduce a typedef since this type is quite long:
```
using llvm::MapVector<Operation *, SmallPtrSet<OpOperand *, 4>> BlockingUsesMap;
```